stytra/stimulation/stimuli/generic_stimuli.py

```diff
@@ -130,7 +143,18 @@ def initialise_external(self, experiment):
             None
 
         """
+        
+        if isinstance(environment_state, EnvironmentState):
+            self._environment_state = environment_state
+        else:
+            self._environment_state = experiment
+            warnings.warn("Warning: 'initialise_external' will use the environment_state variable which holds the calibrator object!", FutureWarning)
+            warnings.warn("Warning: 'initialise_external' will use the environment_state variable which holds the calibrator object!", DeprecationWarning)
```

Comment:
If it's an instance of the EnvironmentState class then initialize the variable correctly, otherwise use the new variable as a place holder for the experiment object 
(only in place to allow backwards compatibility)